sgkit/tests/test_popgen.py

```diff
@@ -68,18 +69,51 @@ def test_divergence(size, n_cohorts):
     ds = divergence(ds)
     div = ds.stat_divergence.values
 
+    # entries on the diagonal are diversity values
+    for i in range(n_cohorts):
+        ts_div = ts.diversity([subsets[i]], span_normalise=False)
+        np.testing.assert_allclose(div[i, i], ts_div)
+
+    # test off-diagonal entries, by replacing diagonal with NaNs
+    np.fill_diagonal(div, np.nan)
     ts_div = np.full([n_cohorts, n_cohorts], np.nan)
     for i, j in itertools.combinations(range(n_cohorts), 2):
         ts_div[i, j] = ts.divergence([subsets[i], subsets[j]], span_normalise=False)
-        ts_div[j, i] = ts_div[i, j]
+        ts_div[j, i] = ts.divergence([subsets[j], subsets[i]], span_normalise=False)
     np.testing.assert_allclose(div, ts_div)
 
 
+@pytest.mark.parametrize(
+    "size, n_cohorts",
+    [(2, 2), (3, 2), (10, 2), (100, 2)],
```

Comment:
Why is the set of of size / n_cohort pairs so different for the Hudson vs Nei tests?

Reply:
Thanks for taking a look @eric-czech!

Hudson is tested by comparing it to scikit-allel, which only allows pairs of cohorts (populations), whereas Nei is compared to tskit, which allows any number of cohorts (and considers them in pairs). I've added a note to the test saying this.

https://github.com/pystatgen/sgkit/pull/302/commits/7cc649303c26b154b112c338d60af7bf91a9f638